packages/relay/src/lib/eth.ts

```diff
@@ -1209,8 +1209,33 @@ export class EthImpl implements Eth {
         constants.TYPE_CONTRACT,
```

Comment:
This query

```ts
      const result = await this.mirrorNodeClient.resolveEntityType(address, EthImpl.ethGetCode, requestDetails, [
        constants.TYPE_CONTRACT,
        constants.TYPE_TOKEN,
      ]);
```

does not depend on the `blockNumber`. Does that mean that this issue happen with contracts as well? If that's the case, we should fixed it here. And the logic you included to fetch block+timestamp can be reused for both cases, contract and tokens.